test/SILOptimizer/sil_combine.sil

```diff
@@ -2878,8 +2878,8 @@ bb2:
 // CHECK-NEXT: [[TMP:%.*]] = alloc_stack $T
 // CHECK: [[FN:%.*]] = function_ref @generic_callee
 // CHECK-NEXT: copy_addr [[ARG1]] to [initialization] [[TMP]] : $*T
-// CHECK-NEXT: apply [[FN]]<T, T>([[ARG0]], [[TMP]])
 // CHECK-NEXT: destroy_addr [[ARG1]]
+// CHECK-NEXT: apply [[FN]]<T, T>([[ARG0]], [[TMP]])
```

Comment:
In case anyone looks at this and wonders why this changed, it is b/c our (apply (partial_apply)) -> (partial_apply) (apply') uses the dead partial_apply code to clean up. This is disguised by the optimization that gets rid of trivially dead live ranges. I added a version of this test to the sil_combine_array.sil test (see below) that takes advantage of that optimization being disabled in that file. So it is not as cleaned up and the work being done is obvious.